sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java

```diff
@@ -30,22 +30,38 @@
      * Logger tag.
      */
     public static final String TAG = AndroidSentryClientFactory.class.getName();
+
     /**
      * Default Buffer directory name.
      */
     private static final String DEFAULT_BUFFER_DIR = "sentry-buffered-events";
 
     private Context ctx;
 
+    /**
+     * Construct an AndroidSentryClientFactory using the base Context from the specified Android Application.
+     *
+     * @param app Android Application
+     */
+    public AndroidSentryClientFactory(Application app) {
+        Log.d(TAG, "Construction of Android Sentry from Android Application.");
+
+        this.ctx = app.getBaseContext();
+    }
+
     /**
      * Construct an AndroidSentryClientFactory using the specified Android Context.
      *
      * @param ctx Android Context.
      */
     public AndroidSentryClientFactory(Context ctx) {
-        Log.d(TAG, "Construction of Android Sentry.");
+        Log.d(TAG, "Construction of Android Sentry from Android Context.");
+
+        this.ctx = ctx;
+    }
 
-        this.ctx = ctx.getApplicationContext();
+    public Context getAndroidContext() {
```

Comment:
What is the purpose of this method?

Reply:
That needs to be inlined; I was initially going with a different approach and just didn't remove it yet.